### PR TITLE
Clean up top toolbar to use same metrics as block toolbar

### PR DESCRIPTION
**Pull request by richtabor**
## What?
Not huge changes, but adjusts how the top toolbar applies borders so that they're more in line with the block toolbar's metrics. In doing so the amount of space between controls is much more consistent across the toolbar, resulting in a bit more space before the horizontal scroll is engaged. Also tidies up the parent selector and the movers so that they are centered vertically. 

Noticed while working on https://github.com/WordPress/gutenberg/pull/61118. 

## Testing Instructions

1. Open at the site editor.
2. Set the top toolbar view from the editor header more menu. 
3. Select a block. 
4. See changes. 

### Before
CleanShot 2024-04-25 at 19 13 15

### After
CleanShot 2024-04-25 at 19 11 53

### CODE REVIEW

File: packages/editor/src/components/collapsible-block-toolbar/style.scss [hunk 1]
```diff
@@ -28,21 +28,23 @@
...
 		&::after {
 			content: "";
 			width: $border-width;
 			height: $grid-unit-30;
 			background-color: $gray-300;
-			margin-top: $grid-unit-05;
-			margin-left: $grid-unit;
+			top: $grid-unit-05;
+			position: absolute;
+			right: - $border-width;
```
Comment (jasmussen) [inline packages/editor/src/components/collapsible-block-toolbar/style.scss]: In some cases we do `-$border-width`, here there's a space between. No strong opinion on which path is better, but might be good to be consistent?

File: packages/editor/src/components/collapsible-block-toolbar/style.scss [hunk 2]
```diff
@@ -63,7 +65,7 @@
 		@include break-small() {
 			&:not(.is-horizontal) .block-editor-block-mover__move-button-container {
 				position: relative;
-				top: -10px;
+				top: - $grid-unit;
```
Comment (jasmussen) [inline packages/editor/src/components/collapsible-block-toolbar/style.scss]: Same here.

**Review by jasmussen — APPROVED** [review]
Nice, works well for me, I also verified that narrow contexts with always shown scrollbar still works well:
Screenshot 2024-04-26 at 08 34 57

File: packages/editor/src/components/collapsible-block-toolbar/style.scss (hunk 2 above)
Comment (t-hamano) [reply packages/editor/src/components/collapsible-block-toolbar/style.scss]: Increasing this value will shift the toolbar when focusing on the block mover button on Windows OS. I should have left a more detailed comment. See [this comment](https://github.com/WordPress/gutenberg/pull/60729#issuecomment-2058230476) for details. If we can make the header height 64px in #60729, there may be no problem with applying `top: - $grid-unit;`.

https://github.com/WordPress/gutenberg/assets/54422211/ef0a1291-96f4-4fba-89e3-2318367256ed

Comment (richtabor) [reply packages/editor/src/components/collapsible-block-toolbar/style.scss]: If we don't adapt it, it'll stay visually mis-aligned all the time. I suppose it's one + for the 64px header height.

Comment (t-hamano) [reply packages/editor/src/components/collapsible-block-toolbar/style.scss]: > If we don't adapt it, it'll stay visually mis-aligned all the time

Yes, I understand that too. This point has been discussed in #57444, where custom scrollbars were introduced (See [this comment](https://github.com/WordPress/gutenberg/pull/57444#issuecomment-1909695979)). Do you think it's okay to merge #60729 first? After that, we can check again to see if `top: - $grid-unit;` does not cause a shift.

Comment (richtabor) [reply packages/editor/src/components/collapsible-block-toolbar/style.scss]: I'm ok taking out that `top: - $grid-unit;` adjustment for now, to not hold this up.

**Pull request merged**